src/Lifecycle.js

```diff
@@ -436,7 +450,13 @@ export function stopMatrixClient() {
     if (cli) {
         cli.stopClient();
         cli.removeAllListeners();
-        cli.store.deleteAllData();
-        MatrixClientPeg.unset();
     }
+
+    if (clearStores) {
+        // note that we have to do this *after* stopping the client, but
+        // *before* clearing the MatrixClientPeg.
```

Comment:
Oh, because we access `MatrixClientPeg` in `_clearStorage()` rather than just passing in `cli`. Boo. This makes me sad because it means ordering is important (enough for it to be documented like you have done so) simply because we access the global rather than giving it the things it needs to work on. Any reason why we can't pass it in? E.g
```js
function _clearStorage(cli) {
  // ...
}
```

Even that is kinda bleh since there's massive side-effects (clearing `localStorage` and analytics) and that function signature makes it seem like it has everything it needs and won't side-effect, whereas the 0 arg function is more obviously going to side-effect. Ugh. If you want to change it, go for it, if you don't then sure. It all makes me sad.